tests/installation/bootloader_hyperv.pm

```diff
@@ -229,4 +220,97 @@ sub run {
+        last if hyperv_cmd("if exist $root\\cache\\$basenamehdd_vhd ( exit 1 )", {ignore_return_code => 1});
+
+        # Copy HDD from NFS share to local cache on Hyper-V
+        hyperv_cmd_with_retry("copy /z $root_nfs\\$hddpath\\$basenamehdd $root\\cache\\");
+
+        # Decompress the XZ compressed image
+        if ($hdd =~ m/vhdx\.xz/) {
+            record_info 'unxz', "Decompressing $root\\cache\\$basenamehdd";
+            my ($ret, $stdout, $stderr) = $svirt->run_cmd("xz --decompress --keep --verbose $root\\cache\\$basenamehdd", wantarray => 1);
+            defined($stderr) && $stderr =~ /xz: $root\\cache\\$basenamehdd: File exists/ && sleep 60;
+            last;
+        }
+    }
+
+    # Ensure the disk file is present
+    hyperv_cmd("if not exist $root\\cache\\" . $basenamehdd =~ s/vhdx\.xz/vhdx/r . " ( exit 1 )");
+}
+
+sub download_iso_from_http {
+    my ($iso, $root, $iso_base_url, $svirt) = @_;
+    my $basenameiso = basename($iso);
+    my $iso_url = "$iso_base_url/$basenameiso";
+
+    # Check if the file is being used by another job process
+    hyperv_cmd_with_retry("if exist $root\\cache\\$basenameiso ( sha256sum $root\\cache\\$basenameiso )", {msgs => ('Permission denied')});
+
+    # Download ISO from HTTP to local cache on Hyper-V
+    unless (-e "$root\\cache\\$basenameiso") {
+        record_info 'Invoke-WebRequest', "Downloading from $iso_url to $root\\cache\\$basenameiso";
+        my $powershell_cmd = "powershell -Command Invoke-WebRequest -Uri '$iso_url' -OutFile '$root\\cache\\$basenameiso'";
+        my ($ret, $stdout, $stderr) = $svirt->run_cmd($powershell_cmd, wantarray => 1, timeout => 600);
+        if ($ret != 0) {
+            record_info 'Invoke-WebRequest', "Failed to download $iso_url: $stderr";
+        }
+    }
+}
+
+sub download_hdd_from_http {
+    my ($hdd, $root, $hdd_base_url, $svirt) = @_;
+    my $basenamehdd = basename($hdd);
+    my $hdd_url = "$hdd_base_url/$basenamehdd";
+
+    # If the image already exists, do nothing
+    if (-e "$root\\cache\\$basenamehdd") {
+        return;
+    }
+
+    # Download HDD from HTTP to local cache on Hyper-V
+    record_info 'Invoke-WebRequest', "Downloading from $hdd_url to $root\\cache\\$basenamehdd";
+    my $powershell_cmd = "powershell -Command Invoke-WebRequest -Uri '$hdd_url' -OutFile '$root\\cache\\$basenamehdd'";
+    my ($ret, $stdout, $stderr) = $svirt->run_cmd($powershell_cmd, wantarray => 1);
+    if ($ret != 0) {
+        record_info 'Invoke-WebRequest', "Failed to download $hdd_url: $stderr";
+    }
+
+    # If necessary, decompress the XZ compressed image
+    if ($hdd =~ m/vhdx\.xz/) {
+        record_info 'unxz', "Decompressing $root\\cache\\$basenamehdd";
+        my ($ret, $stdout, $stderr) = $svirt->run_cmd("xz --decompress --keep --verbose $root\\cache\\$basenamehdd", wantarray => 1);
+        if (defined($stderr) && $stderr =~ /xz: $root\\cache\\$basenamehdd: File exists/) {
+            sleep 60;
```

Comment:
Why sleep 60 ?

Reply:
the origin code contains 'sleep 60', I can try to remove it.